### ISSUE TITLE
Lint to warn on imports that only resolve because of other imports

### ISSUE THREAD

**Issue by mi2ebi** [rust-lang/rust-clippy#17594]
### What it does

Denies cases like the example below. Sometimes, but not always?, rustc suggests importing things this way, which is misleading and confusing, because the only reason `bar::Foo` exists is that `bar` imports `use foo::Foo::self`.

### Advantage

Makes imports more clear.

### Drawbacks

rustc arguably shouldn't generate this in the first place, so maybe we should just fix that instead.

### Example

```rust
// src/bar.rs
use crate::{
    bar::Foo::Foo2,
    foo::Foo::{self, Foo0, Foo1},
};

const fn bar(x: i32) -> Foo {
    match x {
        0 => Foo0,
        1 => Foo1,
        _ => Foo2,
    }
}

// src/foo.rs
pub enum Foo {
    Foo0,
    Foo1,
    Foo2,
}

// src/lib.rs
mod bar;
mod foo;

```

Could be written as:

```rust
// src/bar.rs
use crate::foo::Foo::{self, Foo0, Foo1, Foo2};

const fn bar(x: i32) -> Foo {
    match x {
        0 => Foo0,
        1 => Foo1,
        _ => Foo2,
    }
}
```

### Comparison with existing lints

I don't think there's a similar lint already

**Comment by mi2ebi** [comment]
An example rustc does suggest this for:

<img width="1079" height="255" alt="Image" src="https://github.com/user-attachments/assets/07c09f4f-9295-42fe-b81b-5d1c347acd55" />

(`Jvofli` really lives in `jvot3/src/jvofli.rs`)
